addons/pos_self_order/tests/test_self_order_common.py

```diff
@@ -50,3 +50,25 @@ def test_self_order_pos_closed(self):
         for mode in ("mobile", "consultation", "kiosk"):
             self.pos_config.write({"self_ordering_mode": mode})
             self.start_tour(self_route, "self_order_pos_is_closed")
+
+    def test_self_order_preparation_disabling_preparation_display(self):
+        """
+        This test ensures that the preparation display option can be disabled when the self_ordering_mode is set to 'nothing'.
+        It also tests that the preparation display option is enabled automatically when the self_ordering_mode is set to 'kiosk'.
+        """
+        self.pos_config.self_ordering_pay_after = 'each'
+
+        with odoo.tests.Form(self.env['res.config.settings']) as form:
+            form.module_pos_preparation_display = False
+
+            self.pos_config.write({
+                'self_ordering_mode': 'nothing',
+            })
+            form.pos_config_id = self.pos_config
+            self.assertEqual(form.module_pos_preparation_display, False)
```

Comment:
Just a comment. It would have also worked directly modifying the form values like `form.pos_self_ordering_mode` but since this also works, I'm good with it.